### DISCUSSION TITLE
Symfony Local Web Server

### DISCUSSION THREAD

**Discussion by yguedidi** [wallabag/wallabag#6807] [Ideas]
Hey!

Working on the upgrade to Symfony 5, I noticed somthing: WebServerBundle is removed in 5.0 and we should use Symfony Local Web Server.
This means installing the Symfony CLI, SSL certificates, setting up the proxy, etc (see [here](https://symfony.com/doc/4.4/setup/symfony_server.html))
Not sure this will fit well inside a Docker container... Or it will be very complex!

So here my proposal to simplify developer experience: have PHP and a configured Symfony CLI required on the host.
I think it's an acceptable compromise, because as PHP developer we are likely to have PHP already installed, and Symfony CLI is a developer tool that can manage multiple projects.

Benefits:
- no more Docker mount for the code
- wallabag.wip local domain in dev
- aligned with Symfony practices
- use standard linux alternatives to switch PHP version
- opportunity to rework the setup documentation

Drawbacks:
- more things to setup locally to have all dependencies (mainly PHP extensions)

What do you think?
A simple idea of what to be done is that we will replace docker/php/Dockerfile by an official node container

**Comment by Kdecherf** [comment]
I don't think setting up TLS and a proxy is mandatory. We could probably use it in the Docker container as well.

I mostly disagree on the fact that PHP developers have a PHP stack on their host as my main development environment now is VSCode Dev containers and it works pretty well in my use case. However that doesn't make it mandatory to use Docker for developing features on this repo.

**Reply by yguedidi** [reply]
Good that I asked then! Thanks for your feedback (and all the reviews and approves!)
I'm curious about others opinion but I understand you point of view :)

**Comment by nicosomb** [comment]
I don’t use docker to develop on wallabag but I understand the @Kdecherf ’s opinion and I agree with that. I think we should keep docker configuration.

**Comment by yguedidi** [comment]
OK I'll try to make it work with Docker them.
FYI, I opened https://github.com/symfony-cli/symfony-cli/issues/336 in case you want to participate or just follow :)

**Comment by yguedidi** [comment]
I managed to make it work in https://github.com/wallabag/wallabag/pull/6850

**Comment by j0k3r** [comment]
I think we do have a different configuration for each developer haha
I have PHP instealled locally with extensions but I use Docker for each database (MySQL & PostgreSQL).

**Reply by yguedidi** [reply]
Indeed haha!
Let's leave optimal dev environment topic for another discussion, for now the focus is on finding a good solution to replace WebServerBundle that is removed in Symfony 5.x :)
And #6850 is already a working solution.
I'll try to think about a dev env solution that fits everyone after we reached Symfony 6 :p
